### ISSUE TITLE
Leaves and allfaces_optional

### ISSUE THREAD

**Issue by ghost** [luanti-org/luanti#10231]
##### Minetest version

```
v5.4-dev
```

##### OS / Hardware
Operating system: Linux

##### Summary

- **Visual_scale = 1.2**
- If the set is specifically "Fancy Leaves", then the leaf node (or texture?) expands. So the leaves overlap in the trees and a flickering effect is produced.

##### Steps to reproduce

[
![imagen](https://user-images.githubusercontent.com/13217027/88483725-14c29180-cf6a-11ea-9ed2-79134ff513d4.png)
](url)

Well i think is not a bug, maybe, but this did not occurs in previous versions.

**Comment by SmallJoker** [comment]
What would you expect Minetest to do instead of overlapping? Use a visual scale of 1.

**Comment by sfan5** [comment]
I'm guessing the reported bug here is that visual_scale behaves inconsistently for drawtype = "allfaces_optional",

**Comment by paramat** [comment]
^ runsy is that what this issue is about, that 'visual_scale' only applies to fancy leaves instead of all 3 leaves types?

The flickering is expected for overlapping textures and is not a bug, if you do not like it use visual_scale = 1.

**Comment by runsy** [comment]
Yes, I put visual_scale =1, **so close this issue**. But the fact is that still this version the visual_scale set was ignored.

**Comment by runsy** [comment]
> 
> 
> ^ runsy is that what this issue is about, that 'visual_scale' only applies to fancy leaves instead of all 3 leaves types?
> 
> The flickering is expected for overlapping textures and is not a bug, if you do not like it use visual_scale = 1.

Yes only for fancy leaves.

**Comment by paramat** [comment]
Ok thanks. No need to close this, it could be considered a request for 'visual_scale' to be applied to all modes of 'allfaces_optional', or at least document the current behaviour.

**Comment by SmallJoker** [comment]
@runsy Does this issue persist after #10225 ?

**Comment by runsy** [comment]
Yes, it persists. I try it now in 5.5 visual_scale with my cool trees mod only applies to 'fancy leaves'.

This bug not important for me, cos I changed  visual_scale to 1.

**Comment by Wuzzy2** [comment]
Why is this considered a documentation issue? Also, please rewrite the issue text to contain the actual bug, it's confusing.

**Comment by Wuzzy2** [comment]
This issue is now outdated and should probably be closed. `visual_scale != 1` is no longer officially supported for `allfaces_optional`, as of Luanti 5.17.0.

`lua_api.md`:

    visual_scale = 1.0,
    -- Supported for drawtypes "plantlike", "signlike", "torchlike",
    -- "firelike", "mesh", "nodebox", "allfaces". 
    -- For plantlike and firelike, the image will start at the bottom of the
    -- node. For torchlike, the image will start at the surface to which the
    -- node "attaches". For the other drawtypes the image will be centered
    -- on the node.
    -- note: Do not set a visual_scale != 1.0 for drawtypes that don't support it.

Also, it's not clear what OP wanted anyway (and the account is deleted ("ghost" = deleted account)).

**Comment by SmallJoker** [comment]
Additionally fixed by #17325.
